docs/loadtesting.md

```diff
@@ -0,0 +1,42 @@
+# Load Testing
+
+We want to observe how the server behaves under realistic load. To
+keep things realistic, our approach should:
+
+- use real HTTPS requests against a load-testing system configured similarly to the real system
+- use requests that look as much as possible like the real thing, e.g. with expected cookies
+- figure out how to load test Arlo without also load-testing Auth0 for authentication
+- whatever hacks we put in for authentication (it should be clear by
+  now we'll need some), they should not affect production code.
+- scale up well with ideally minimal infrastructural investment so
+  we're not manually spinning up dozens of machines ourselves.
+
+## Authentication Hack for Load Testing
+
+We have superadmin features that allow for impersonation of any user,
+and which uses the same cookies as normal logins. Let's reuse that.
+
+We'll build a superadmin bypass using a special HTTP header. Because
+we don't ever want that code in production, we'll keep that code in a
+special branch, `loadtesting`, that we'll need to rebase every so
+often on `master`. We'll also require a special environment variable
+be set as an extra defense-in-depth measure, just in case we
+mistakenly merge that branch into `master`. Belt _and_ suspenders.
```

Comment:
we're gonna need this for our integration tests as well, so i think we'll need something we feel comfortable having on `master`

this overall approach seems mostly fine to me. my small concern is overloading the superadmin feature - it might need to change for other reasons and then we'd have to worry about preserving this

i was wondering if there's any way to mock auth0 to make this more straightforward. we could still do defense-in-depth

Reply:
Mocking auth0 in a running server? That is tricky and dangerous, effectively implementing another auth bypass in addition to superadmins, I don't think that's great.